drivers/docker/driver.go

```diff
@@ -1078,6 +1078,9 @@ func (d *Driver) createContainerConfig(task *drivers.TaskConfig, driverConfig *T
 
 	default:
 		if len(driverConfig.PortMap) > 0 {
+			if task.Resources.Ports != nil {
+				return c, fmt.Errorf("'port_map' cannot map group network ports, use 'ports' instead")
```

Comment:
How will this clause be triggered?  It seems it's captured in the first switch case statement in line 1055 above:

```go
	case task.Resources.Ports != nil && len(driverConfig.Ports) > 0:
```

Reply:
this clause is triggered if `port_map { ... }` is defined in the driver config (driverConfig.**PortMap**) and group network ports are defined (task.Resources.Ports)

The first case statement is hit if group network ports are defined and `ports = [...]` is defined in the driver config (driverconfig.**Ports**).

Its meant to catch the following configuration:

```hcl
group {
  network {
    port "http" { }
  }
  task {
    driver = "docker"
    config {
      port_map {
        # CANNOT reference group ports here, must use ports = ["http"] and set port.to = 8080
        http = 8080
      }
    }
  }
}
```

Reply:
Makes sense - thanks!